datadog/resource_datadog_powerpack.go

```diff
@@ -0,0 +1,496 @@
+		}
+		if v, ok := templateVariable.GetDefaultsOk(); ok && len(*v) > 0 {
+			var tags []string
+			tags = append(tags, *v...)
+			terraformTemplateVariable["defaults"] = tags
+		}
+		terraformTemplateVariables[i] = terraformTemplateVariable
+	}
+	return &terraformTemplateVariables
+}
+func getPowerpackTemplateVariableSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The name of the powerpack template variable.",
+		},
+		"defaults": {
+			Type:     schema.TypeList,
+			Optional: true,
+			Elem: &schema.Schema{
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			Description: "One or many default values for powerpack template variables on load. If more than one default is specified, they will be unioned together with `OR`.",
+		},
+	}
+}
+
+func resourceDatadogPowerpackCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConf := meta.(*ProviderConfiguration)
+	apiInstances := providerConf.DatadogApiInstances
+	auth := providerConf.Auth
+	powerpackPayload, diags := buildDatadogPowerpack(ctx, d)
+	if diags.HasError() {
+		return diags
+	}
+	powerpack, httpresp, err := apiInstances.GetPowerpackApiV2().CreatePowerpack(auth, *powerpackPayload)
+	if err != nil {
+		return utils.TranslateClientErrorDiag(err, httpresp, "error creating powerpack")
+	}
+	if err := utils.CheckForUnparsed(powerpack); err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(*powerpack.Data.Id)
+
+	var getPowerpackResponse datadogV2.PowerpackResponse
+	var httpResponse *http.Response
+
+	err = retry.RetryContext(ctx, d.Timeout(schema.TimeoutCreate), func() *retry.RetryError {
+		getPowerpackResponse, httpResponse, err = apiInstances.GetPowerpackApiV2().GetPowerpack(auth, *powerpack.Data.Id)
+
+		if err != nil {
+			if httpResponse != nil {
+				return retry.RetryableError(fmt.Errorf("powerpack not created yet"))
+			}
+			return retry.NonRetryableError(err)
+		}
+
+		if err := utils.CheckForUnparsed(getPowerpackResponse); err != nil {
+			return retry.NonRetryableError(err)
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return updatePowerpackState(d, &getPowerpackResponse)
+}
+
+func resourceDatadogPowerpackUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConf := meta.(*ProviderConfiguration)
+	apiInstances := providerConf.DatadogApiInstances
+	auth := providerConf.Auth
+	id := d.Id()
+	powerpack, diags := buildDatadogPowerpack(ctx, d)
+	if diags.HasError() {
+		return diags
+	}
+
+	updatedPowerpackResponse, httpResponse, err := apiInstances.GetPowerpackApiV2().UpdatePowerpack(auth, id, *powerpack)
+	if err != nil {
+		if httpResponse != nil {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  fmt.Sprintf("error updating powerpack: %s", err),
+			})
```

Comment:
```suggestion
			diags = append(diags, utils.TranslateClientErrorDiag(err, httpResponse, "error updating powerpack")...)
```
we should use the http reponse in the error message in case the api returns a useful error message, in particular for layout validation